feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ConversationAnswerMessagesUITest.kt

```diff
@@ -0,0 +1,312 @@
+                onCreatePost = { CompletableDeferred() },
+                onEditPost = { _, _ -> CompletableDeferred() },
+                onResolvePost = { CompletableDeferred() },
+                onDeletePost = { CompletableDeferred() },
+                onRequestReactWithEmoji = { _, _, _ -> CompletableDeferred() },
+                onRequestReload = {},
+                onRequestRetrySend = { _, _ -> },
+            )
+        }
+
+        composeTestRule.onNodeWithText("Post content").assertExists()
+        composeTestRule.onNodeWithText("Answer Post content 0").assertExists()
+        composeTestRule.onNodeWithText("Answer Post content 1").assertExists()
+        composeTestRule.onNodeWithText("Answer Post content 2").assertExists()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_is_resolved))
+            .assertDoesNotExist()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_resolves))
+            .assertDoesNotExist()
+    }
+
+    @Test
+    fun `test GIVEN the post is resolved and one answer post is marked as resolving THEN the post is shown as resolved and this answer post is shown as resolving`() {
+        var resolvingIndex = 0
+        val resolvedPost = post.copy(
+            resolved = true,
+            answers = post.answers.mapIndexed { index, answer ->
+                resolvingIndex = post.answers.indexOfFirst { !it.resolvesPost }
+                if (!answer.resolvesPost && index == resolvingIndex) {
+                    // Update only the first unresolved answer
+                    answer.copy(resolvesPost = true)
+                } else {
+                    answer
+                }
+            }
+        )
+
+        composeTestRule.setContent {
+            MetisThreadUi(
+                modifier = Modifier.fillMaxSize(),
+                courseId = course.id!!,
+                clientId = clientId,
+                postDataState = DataState.Success(resolvedPost),
+                conversationDataState = DataState.Success(conversation),
+                hasModerationRights = false,
+                isAtLeastTutorInCourse = false,
+                serverUrl = "",
+                initialReplyTextProvider = remember { TestInitialReplyTextProvider() },
+                onCreatePost = { CompletableDeferred() },
+                onEditPost = { _, _ -> CompletableDeferred() },
+                onResolvePost = { CompletableDeferred() },
+                onDeletePost = { CompletableDeferred() },
+                onRequestReactWithEmoji = { _, _, _ -> CompletableDeferred() },
+                onRequestReload = {},
+                onRequestRetrySend = { _, _ -> },
+            )
+        }
+
+        composeTestRule.onNodeWithText("Post content").assertExists()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_is_resolved)).assertExists()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_resolves)).assertExists()
```

Comment:
This misses a check that this string is actually shown on the correct answer post.